pkg/webhook/shoot/mutator.go

```diff
@@ -76,3 +85,41 @@ func (m *mutator) mutateVPNShootDeployment(_ context.Context, deployment *appsv1
 
 	return nil
 }
+
+func (m *mutator) mutateCalicoNode(_ context.Context, ds *appsv1.DaemonSet) error {
+	if c := extensionswebhook.ContainerWithName(ds.Spec.Template.Spec.Containers, "calico-node"); c != nil {
+		ebpfEnabled := slices.ContainsFunc(c.Env, func(e corev1.EnvVar) bool {
+			if e.Name != "FELIX_BPFENABLED" {
+				return false
+			}
+
+			enabled, _ := strconv.ParseBool(e.Value)
+
+			return enabled
+		})
+
+		if !ebpfEnabled {
+			return nil
+		}
+
+		m.logger.Info("patching calico-node daemon set due to ebpf dataplane being enabled")
+
+		c.Env = extensionswebhook.EnsureEnvVarWithName(c.Env, corev1.EnvVar{
+			Name: "FELIX_BPFDATAIFACEPATTERN",
+			// including "lan" interface name to default value
```

Comment:
can you please point to the original source of this regex horror

Reply:
Added the reference.